### ISSUE TITLE
Tips for positioning far apart objects of different sizes

### ISSUE THREAD

**Issue by manoaman** [google/neuroglancer#311]
Hi,

I have a use case where I have a mesh object (1mm^3), and a skeleton object (1um^3), which are far apart and not visible in the default 3D view.  Either one or the other could be seen from adjusting the scales (zoom-in/zoom-out level) and the center position.  Not both objects in the same view.  If I want to position this very small skeleton object inside a mesh object, let's say position at (0,0,0) so that both objects are visible, what are the UI control tips for repositioning these objects?  The goal would be to put a tiny skeleton inside a big mesh object.

One approach I thought might be modifying the JSON state to position to [0,0,0], and then move smaller object.  Although, it seems like both objects' positions are synched and they don't overlay.  I used Shift+Mouse-drag for UI control.  Are there ways to position all objects in a specified spatial position from GUI?  Any tips?

Thank you,
-m

**Comment by jbms** [comment]
If the objects are from different data sources, you can affect their relative positions by adjusting the affine coordinate transform matrix on the sources tab.  (Especially the translation column).

Unfortunately there is currently no UI control in neuroglancer for e.g. adjusting the translation by dragging the mouse.

**Comment by manoaman** [comment]
Hi @jbms 

Yes, translation does help.  However, the problem I'm having is that roughly I have a skeleton in 1nm^3, and trying to view that very tiny object within 10um^3 mesh.  A skeleton is so tiny to identify even after I adjust the position.

Would it be possible to scale up a skeleton object to 1um^3 from a transformation matrix?  Or does annotation tool let draw a cube around a skeleton big enough to identify so that I can identify it from the bigger mesh object layer?

Thanks,
-m

**Comment by jbms** [comment]
Yes you can use the affine transformation matrix to rescale as well.  You can also rescale by modifying the scale/units listed under "Source dimensions".  Depending on your use case that might or might not be more convenient than modifying the transformation matrix.

**Comment by manoaman** [comment]
I see. After multiple attempts to the transformation matrix, and adjusting translation values, it does help to get both objects close enough in a single view.  One other question.  If I need to rotate just a skeleton on a Y-axis, perhaps tilt 90 degrees angle, how do I use this matrix?  Keyboard R and E seem to work on both objects.  Is there a way to individually rotate?

**Comment by jbms** [comment]
Unfortunately there are no UI controls to adjust the coordinate transformation by mouse movements --- you have to type in numbers.

You can perform rotations, but it is indeed challenging to modify an affine transformation matrix by hand.

Currently if you want them overlaid in the same view there is no other option than the transformation matrix.

If you are happy to have them displayed side-by-side, you create a tiled view in Neuroglancer, by dragging and dropping one of the layers (use the tab in the layer bar) to the side of the view.

This will create a tiled view with two layer groups.  By default the position, orientation, and zoom level of both layer groups will be linked to the global position, orientation, and zoom level.  But if you right click on the blank space next to the layer tabs in one of the layer groups, you can change these properties to be unlinked, or linked with an offset.

**Comment by manoaman** [comment]
Sorry, I'm a bet lost on how to use transformation matrix to perform rotations.  Would you be able to give me an example?  (Changing the size and position, I figured ok.)

In fact, this tiled view with layer groups is a pretty cool feature.  I'm glad I found out this feature.  Thanks for this @jbms !!
